src/ai/ai_gui.cpp

```diff
@@ -223,7 +255,7 @@ struct AIListWindow : public Window {
 	 */
 	virtual void OnInvalidateData(int data = 0, bool gui_scope = true)
 	{
-		if (_game_mode == GM_NORMAL && Company::IsValidID(this->slot)) {
+		if (_game_mode == GM_NORMAL && Company::IsValidAiID(this->slot)) {
```

Comment:
AI settings won't close on invalidation if the company is not an AI?

Reply:
When a company starts, this is called. It is okay to keep it open if it's a human company. But it can be trouble if it's an ai company, because by leaving it open, I could select another AI, which in turn would change the list of AI settings for the slot. The just started AI would be working with a config that wasn't of itself, a recipe for trouble.